src/LatteContext/Collector/VariableCollector.php

```diff
@@ -51,13 +56,37 @@ public function collectData(Node $node, Scope $scope): ?array
             return null;
         }
 
+        if ($this->lattePhpDocResolver->resolveForNode($node, $scope)->isIgnored()) {
+            return null;
+        }
+
+        $isVariablesNode = false;
         $collectedVariables = [];
         foreach ($this->variableCollectors as $variableCollector) {
             if (!$variableCollector->isSupported($node)) {
                 continue;
             }
-            $collectedVariables[] = $variableCollector->collect($node, $scope);
+            $variables = $variableCollector->collect($node, $scope);
+            if ($variables === null) {
+                continue;
+            }
+            $isVariablesNode = true;
+            foreach ($variables as $variable) {
+                $collectedVariables[$variable->getVariableName()] = $variable;
```

Comment:
this will override existing collected variables, it is correct?

Reply:
I can write any collector for example for Class_ node and then iterate over stmts and find some variables (just example).

Reply:
I don't need multiple collectors, My one collector can collect variables of same name multiple times, e.g. in case of I collect something like this with ClassMethod Node:
```
if ($condition) {
    $this->template->a = 'a';
} else {
    $this->template->a = 'b';
}
```
Variable `a` will be represented just with value `'b'` and not `'a'|'b'`

Reply:
I see. Ok I will change it to  union